### PR TITLE
Dasd 8839: Create shared worker app service plan and shared worker app subnet

### PR DESCRIPTION
**Pull request by jack-cloud-platform**
Used for resource compute optimisation for:
- Function Apps with solely Timer triggered functions
- Web Apps with solely schedule triggered WebJobs

### CODE REVIEW

File: templates/subscription.template.json [hunk 1]
```diff
@@ -171,6 +171,15 @@
         "environmentVariable": "apimEndpointSubnetCount"
       }
     },
+    "sharedWorkerSubnetCount": {
+      "type": "int",
+      "defaultValue": 0,
```
Comment (NickGraham101) [inline templates/subscription.template.json:176]: Why is the default value different to environment.template.json?

File: tests/resources/mock.template.json [hunk 2]
```diff
@@ -186,6 +186,15 @@
         "environmentVariable": "apimEndpointSubnetCount"
       }
     },
+    "sharedWorkerSubnetCount": {
+      "type": "int",
+      "defaultValue": 0,
```
Comment (NickGraham101) [inline tests/resources/mock.template.json:191]: Default value here is also different to environment.template.json

File: templates/subscription.template.json (hunk 1 above)
Comment (jack-cloud-platform) [reply templates/subscription.template.json]: Set to 1 along with other subnet count parameters

File: tests/resources/mock.template.json (hunk 2 above)
Comment (jack-cloud-platform) [reply tests/resources/mock.template.json]: Set to 1 along with other subnet count parameters

**Review by NickGraham101 — APPROVED** [review]

**Pull request merged**